xdsl/printer.py

```diff
@@ -142,44 +144,51 @@ def print(self, *argv: Any) -> None:
             text = str(arg)
             self.print_string(text)
 
+    @deprecated("Please use `printer.print_strint(text, indent=0)`")
     def print_string_raw(self, text: str) -> None:
         """
         Prints a string to the printer's output, without taking
         indentation into account.
         """
-        lines = text.split("\n")
-        if len(lines) != 1:
-            self._current_line += len(lines) - 1
-            self._current_column = len(lines[-1])
-        else:
-            self._current_column += len(lines[-1])
-        print(text, end="", file=self.stream)
+        self.print_string(text, indent=0)
 
-    def print_string(self, text: str, indent: int | None = None) -> None:
+    def print_string(self, text: str, *, indent: int | None = None) -> None:
         """
         Prints a string to the printer's output.
 
         This function takes into account indentation level when
         printing new lines.
-
-        Indentation level for newlines within the printed string
-        can be overriden by specifying the `indent` argument.
+        If the indentation level is specified as 0, the string is printed as-is, if `None`
+        then the `Printer` instance's indentation level is used.
         """
 
+        num_newlines = text.count("\n")
```

Comment:
Is this more or less the same speed as a precomputed regex?

Reply:
I'm not sure, but I'll probably fiddle with this only if I see it coming up in the traces

Reply:
It doesn't really make a difference:

```py
# a.py
import random
import time
random.seed(9999)
# text with a few instances of `#b` in there
alphabet = list("abcdefghijklmopqrstuvwxyzABCDEFGHIJKLMOPQRSTUVWXYZ0123456789 ") + ['#b']
text = "".join(
    random.choice(alphabet) for _ in range(500)
)
```

**`str.count`**:

```bash
$ python3 -m timeit -s "from a import text, re" "text.count('#b')"
10000 loops, best of 5: 34 usec per loop
```

**`re.findall` (with compilation)**

```bash
$ python3 -m timeit -s "from a import text, re" "len(re.findall('#b', text))"
10000 loops, best of 5: 33.8 usec per loop
```

**`re.findall` (precompiled)**
```bash
$ python3 -m timeit -s "from a import text, re; expr = re.compile('#b')" "len(expr.findall(text))"
10000 loops, best of 5: 33.7 usec per loop
```

These are all well within error of each other.

Smaller strings seem to even favour `str.count`, probably because the regex machinery is actaully somewhat heavy...